core/coreunix/add.go

```diff
@@ -105,7 +105,11 @@ func (adder *Adder) add(reader io.Reader) (ipld.Node, error) {
 	}
 
 	// Make sure all added nodes are written when done.
-	defer adder.bufferedDS.Commit()
+	defer func() {
+		if err == nil {
+			err = adder.bufferedDS.Commit()
+		}
+	}()
```

Comment:
Cause of https://github.com/ipfs/go-ipfs/issues/6139#issue-425982186. We're _ignoring_ the add error for small files.

Reply:
This will need a named `err` return value or something. Right now it references the first declared `err` and doesn't intercept any of the ones down below.